ignis/services/power_profiles/service.py

```
@@ -0,0 +1,158 @@
+    ) -> str:
+        """
+        Current active power profile.
+
+        Should be either of:
+            - `performance`
+            - `balanced`
+            - `power-saver`
+        """
+        if not self.is_available:
+            raise PowerProfilesDaemonNotRunningError()
+        return self._active_profile
+
+    @active_profile.setter
+    def active_profile(
+        self,
+        profile: str,
+    ) -> None:
+        if not self.is_available:
+            raise PowerProfilesDaemonNotRunningError()
+        if profile not in self.profiles:
+            raise ValueError(f"Profile '{profile}' is not available.")
+
+        self._cookie = -1
+        self._proxy.ActiveProfile = GLib.Variant("s", profile)
+
+    def hold_profile(self, profile: str) -> None:
+        """
+        This forces the passed profile (only `performance` or `power-saver`) to be activated until ignis exits,
+        :func:`~release_profile` is called, or the :attr:`~active_profile` is changed manually.
+
+        Use if you need to ensure a specific profile is active for a certain amount of time or while
+        a specific task is being performed. This way the previous state will not have to be managed by you.
+        """
+        if profile not in self.profiles:
+            raise ValueError(f"Profile '{profile}' is not available.")
+        if profile == "balanced":
+            raise ValueError(
+                "Cannot hold the balanced profile, only performance or power-saver."
+            )
+
+        if not self.is_available:
+            raise PowerProfilesDaemonNotRunningError()
+
+        if self._cookie != -1:
+            return
+
+        self._cookie = self._proxy.gproxy.HoldProfile(
+            "(sss)", profile, "", "com.github.linkfrg.ignis"
+        )
+
+    def release_profile(self) -> None:
+        """
+        Release the hold on the profile
+        """
+        if not self.is_available:
+            raise PowerProfilesDaemonNotRunningError()
```

Okay, as a final change I can suggest, let's move this repeated pattern into a simple function:
```
def __check_available(self) -> None:
    if not self.is_available:
        raise PowerProfilesDaemonNotRunningError()
```

---

> I'd say this is kinda pointless. You're abstracting 2 lines of code from a couple of methods to gain back literally 1 line.

I could say these two lines are pretty bulky, especially because of the long name of the exception. Just calling `self.__check_available()` feels a lot cleaner.


> If at some point one of these methods that calls `__check_available` will need to handle `not self.is_available` in a different way it will create more work then it should've

You're telling about `profiles`? I think we should just raise the exception here (use `__check_available`). And in case of possible future functionality that requires a different way to handle `not self.is_available`, we can just leave it as is. `__check_available` mainly targets functions which should raise the exception, all others should have their own handlers.

---

Alright, your point of view sounds convincing. Let's leave the things as they are